assets/js/resetvalidate.js

```diff
@@ -0,0 +1,168 @@
+$(function () {
+  $('#mail').keyup(function() {
+    var input = $(this).val();
+
+    if (input != "") {
+      $.ajax ({
+
+        url:'/reset/authenticate',
+        method:'POST',
+        data:{mail:input},
+
+        success:function(data) {
+          if (data) {
+            $('#mail-error').html(data);
+            $('#mail-error').css("display", "block");
+            $('#submit').prop("disabled", true);
+          } else {
+            $('#mail-error').html(data);
+            $('#mail-error').css("display", "block");
+            $('#submit').prop("disabled", false);
+          }
+        }
+      });
+    } else {
+      $('#mail-error').css("display", "none");
+    }
+  });
+
+  $('#submit').on('click', function() {
+    var input = $('#mail').val();
+    if (input != "") {
+      $('#loading').show();
+      $.ajax ({
+        url:'/reset/action',
+        method:'POST',
+        data:{email:input},
+        success:function(data) {
+          if (data) {
+            $('#otpbox').modal('show');
+            $('#loading').hide();
+            $('#resend').prop("disabled", true);
+          }
+        }
+      });
+    } else {
+      $('#mail-error').html("This field cannot be empty!");
+    }
+  });
+
+  $('#pin4').keyup(function() {
+
+    var input = $('#pin1').val() + '' + $('#pin2').val() + '' + $('#pin3').val() + '' + $('#pin4').val();
```

Comment:
What is this code doing?

Reply:
In this piece of code, the input provided by the user in four input boxes is concatenated into a single value which is validated with the database value.